src/tests/ftest/dfuse/container_attrs.py

```diff
@@ -0,0 +1,113 @@
+"""
+  (C) Copyright 2020-2024 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import os
+import re
+
+from apricot import TestWithServers
+from dfuse_utils import get_dfuse, start_dfuse
+
+
+class DfuseContainerAttrs(TestWithServers):
+    """Check if the dfuse attributes of a container are properly managed.
+
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up each test case."""
+        # obtain separate logs
+        self.update_log_file_names()
+
+        # Start the servers and agents
+        super().setUp()
+
+    def _check_attrs(self, dfuse, container_name):
+        """Check if the DFuse attributes of a container are loaded
+
+        Check in the log file of the dfuse instance if it contains the DFuse attributes of a given
+        container.  It also checks the value of the attributes found.
+
+        Args:
+            dfuse (Dfuse): DFuse instance to check
+            container_name (str): Name of the container
+        """
+        log_file = dfuse.get_log_file()
+        attrs = self.params.get("attrs", f"/run/{container_name}/*")
+        for attr in [attr.split(':') for attr in attrs.split(",")]:
+            match = None
+            attr_re = re.compile(r"^.+\ssetting\s+'" + attr[0] + r"'\s+is\s+(\d+)\s+seconds$")
+            for line in log_file:
+                match = attr_re.match(line)
```

Comment:
Would a `re.findall(r"^.+\ssetting\s+'" + attr[0] + r"'\s+is\s+(\d+)\s+seconds$", logfile)` be more efficient here?

Reply:
Not sure to understand as findall will not work with the logfile in parameter.
For me, as for match, findall is a one line regexp function.
> [re.match](https://docs.python.org/2/library/re.html#re.match) matches the pattern from the start of the string.  [re.findall](https://docs.python.org/2/library/re.html#re.findall) however searches for occurrences of the pattern anywhere in the string.
As we are using regex matching the whole line, I do not think we will some significant performance difference with match() or findall() function

Reply:
It is up to you, but you can find all the matches with re.findall(), e.g.
```
>>> print(log_file)
10/21-09:48:29.41 wolf-66 DAOS[279335/279341/0] dfuse INFO src/client/dfuse/dfuse_core.c:707 dfuse_cont_get_cache(0x7f2b2c012ee0) setting 'dfuse-attr-time' is 42 seconds
10/21-09:48:29.41 wolf-66 DAOS[279335/279341/0] dfuse INFO src/client/dfuse/dfuse_core.c:707 dfuse_cont_get_cache(0x7f2b2c012ee0) setting 'dfuse-dentry-time' is 404 seconds
10/21-09:48:29.41 wolf-66 DAOS[279335/279341/0] dfuse INFO src/client/dfuse/dfuse_core.c:707 dfuse_cont_get_cache(0x7f2b2c012ee0) setting 'dfuse-attr-time' is 43 seconds
10/21-09:48:29.41 wolf-66 DAOS[279335/279341/0] dfuse INFO src/client/dfuse/dfuse_core.c:707 dfuse_cont_get_cache(0x7f2b2c012ee0) setting 'dfuse-dentry-time' is 405 seconds
10/21-09:48:29.41 wolf-66 DAOS[279335/279341/0] dfuse INFO src/client/dfuse/dfuse_core.c:707 dfuse_cont_get_cache(0x7f2b2c012ee0) setting 'dfuse-attr-time' is 44 seconds
10/21-09:48:29.41 wolf-66 DAOS[279335/279341/0] dfuse INFO src/client/dfuse/dfuse_core.c:707 dfuse_cont_get_cache(0x7f2b2c012ee0) setting 'dfuse-dentry-time' is 406 seconds
10/21-09:48:29.41 wolf-66 DAOS[279335/279341/0] dfuse INFO src/client/dfuse/dfuse_core.c:707 dfuse_cont_get_cache(0x7f2b2c012ee0) setting 'dfuse-attr-time' is 45 seconds
10/21-09:48:29.41 wolf-66 DAOS[279335/279341/0] dfuse INFO src/client/dfuse/dfuse_core.c:707 dfuse_cont_get_cache(0x7f2b2c012ee0) setting 'dfuse-dentry-time' is 407 seconds
>>> 
>>> attr = 'dfuse-dentry-time'
>>> re.findall(r".+\ssetting\s+'" + attr+ r"'\s+is\s+(\d+)\s+seconds", log_file)   
['404', '405', '406', '407']
>>> 
>>> attr = 'dfuse-attr-time'
>>> re.findall(r".+\ssetting\s+'" + attr+ r"'\s+is\s+(\d+)\s+seconds", log_file)
['42', '43', '44', '45']
>>> 
```

Or with a single re.findall(), e.g.
```
>>> attrs = 'dfuse-dentry-time:405,dfuse-attr-time:45'
>>> attr_keys = [attr.split(':')[0] for attr in attrs.split(",")] 
>>> re.findall(fr".+\ssetting\s+'({'|'.join(attr_keys)})'\s+is\s+(\d+)\s+seconds", log_file) 
[('dfuse-attr-time', '42'), ('dfuse-dentry-time', '404'), ('dfuse-attr-time', '43'), ('dfuse-dentry-time', '405'), ('dfuse-attr-time', '44'), ('dfuse-dentry-time', '406'), ('dfuse-attr-time', '45'), ('dfuse-dentry-time', '407')]
>>>
```

Reply:
- Use of `findall()` instead of loop

Reply:
> * [x]  Use of `findall()` instead of loop

Fixed with commit 2e9afbd28